### PR TITLE
hotfix for pyam aggregation and filter in the wild

### PR DESCRIPTION
**Pull request by gidden**
@zikolach has run into a few issues when trying to use pyam in the wild with unruly datasets. This hotfix patches some issues seen when certain functions are called with missing data. Specifically:

- `df.aggregate_region()` now works as expected when the region-to-aggregate-to DNE in the dataframe already
- `df.filter()` now does not error if `df` is empty

There is also an example test that can be implemented when `aggregate_region()` is updated as referenced in #255

### CODE REVIEW

**Comment by gidden** [comment]
Interestingly, this passes on my laptop, but not on my desktop...

**Comment by danielhuppmann** [comment]
the new test fails on my laptop with
```
        pd.testing.assert_frame_equal(obs.reset_index(drop=True),
>                                     exp.reset_index(drop=True))
E       AttributeError: 'memoryview' object has no attribute 'reset_index'
```
which is different from the error seen in CI. The offending line in CI passes.

**Comment by gidden** [comment]
As requested @danielhuppmann, each commit is self contained:

- fa93e4b shows the original failure seen by @zikolach 
- 2e835b8 fixes that error
- 6a812a7 fixes another related issue seen by @zikolach in the wild when trying to filter empty dataframes
- c6685a7 provides two additional tests to confirm that behavior is now as @zikolach expects when doing region aggregation

File: tests/test_core.py [hunk 1]
```diff
@@ -182,6 +182,15 @@ def test_variable_unit(test_df):
     npt.assert_array_equal(test_df.variables(include_units=True), exp)
 
 
+def test_filter_empty_df():
+    # test for issue seen in #254
+    cols = ['model', 'scenario', 'region', 'variable', 'unit', 2005, 2010]
```
Comment (danielhuppmann) [inline tests/test_core.py]: suggest to use `pyam.IAMC_IDX + [2005, 2010]`

File: tests/test_feature_aggregate.py [hunk 2]
```diff
@@ -1,9 +1,56 @@
...
+    # obs = check_aggregate_df.filter(region='foo').data
+    # assert len(exp) > 0
+    # pd.testing.assert_frame_equal(obs.reset_index(drop=True),
+    #                               exp.reset_index(drop=True))
+
+
+def test_aggregate_region_extra_subregion():
+    cols = ['model', 'scenario', 'region', 'variable', 'unit', 2005, 2010]
+    data = pd.DataFrame([
+        ['TEST', 'scen', 'China', 'Primary Energy', 'EJ/y', 1, 6],
```
Comment (danielhuppmann) [inline tests/test_feature_aggregate.py]: please change model and scenario for consistency with `pyam/tests/conftest:TEST_DF`

File: tests/test_feature_aggregate.py [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+                              region='R5ASIA',
+                              subregions=['China', 'Vietnam', 'Japan'],
+                              components=[], append=False)
+    assert len(obs) == 2
+
+
+def test_aggregate_region_missing_all_subregions():
+    cols = ['model', 'scenario', 'region', 'variable', 'unit', 2005, 2010]
+    data = pd.DataFrame([
+        ['TEST', 'scen', 'foo', 'Primary Energy', 'EJ/y', 1, 6],
```
Comment (danielhuppmann) [inline tests/test_feature_aggregate.py]: as above, please change model and scenario for consistency with `pyam/tests/conftest:TEST_DF`

**Comment by zikolach** [comment]
@gidden can you please update PR according to @danielhuppmann? Or push it to origin and let me know so I can do it

**Comment by gidden** [comment]
Hi @zikolach, feel free to just PR into my remote. Once I pull it will be updated here as well.

File: tests/test_core.py (hunk 1 above)
Comment (zikolach) [reply tests/test_core.py]: fixed

File: tests/test_feature_aggregate.py (hunk 2 above)
Comment (zikolach) [reply tests/test_feature_aggregate.py]: fixed

File: tests/test_feature_aggregate.py (hunk 3 above)
Comment (zikolach) [reply tests/test_feature_aggregate.py]: fixed

**Comment by zikolach** [comment]
@gidden I pushed a commit implementing @danielhuppmann suggestions. On my machine tests pass. Please rebase this PR as it has (trivial) conflicting changes in README.

**Comment by gidden** [comment]
Ok, fixed the merge conflict here. @danielhuppmann if this meets your approval, I think it should be g2g after CI.

**Pull request merged**